src/components/editor/Bindings/SchemaInference/Button.tsx

```diff
@@ -53,36 +53,34 @@ function SchemaInferenceButton() {
             )
                 .then(
                     (response) => {
+                        let inferredSpec = null;
                         if (Object.hasOwn(collectionData.spec, 'writeSchema')) {
-                            const { writeSchema, ...additionalSpecKeys } =
+                            const { ...additionalSpecKeys } =
                                 collectionData.spec;
```

Comment:
No longer need to remove the `writeSchema` here. Another fix could have been to also remove `readSchema` here as well.